File: src/test/java/com/kobylynskyi/graphql/codegen/scala/compile-pass/EventPropertyParentParametrizedInput.scala
import com.kobylynskyi.graphql.codegen.model.graphql.GraphQLParametrizedInput
import com.kobylynskyi.graphql.codegen.model.graphql.GraphQLRequestSerializer
import scala.collection.JavaConverters._
import EventStatus._

/**
 * Parametrized input for field parent in type EventProperty
 */
case class EventPropertyParentParametrizedInput(
    withStatus: EventStatus,
    createdAfter: String
) extends GraphQLParametrizedInput {

    override def deepCopy(): EventPropertyParentParametrizedInput = {
        EventPropertyParentParametrizedInput(
            this.withStatus,
            this.createdAfter
        )
    }

    override def toString(): String = {
        scala.Seq(
            if (withStatus != null) "withStatus: " + GraphQLRequestSerializer.getEntry(withStatus) else "",
            if (createdAfter != null) "createdAfter: " + GraphQLRequestSerializer.getEntry(createdAfter) else ""
        ).filter(_ != "").mkString("(", ",", ")")
    }

}

File: src/test/java/com/kobylynskyi/graphql/codegen/kotlin/compile-pass/EventPropertyParentParametrizedInput.kt
import com.kobylynskyi.graphql.codegen.model.graphql.GraphQLParametrizedInput
import com.kobylynskyi.graphql.codegen.model.graphql.GraphQLRequestSerializer
import java.util.StringJoiner
/**
 * Parametrized input for field parent in type EventProperty
 */
data class EventPropertyParentParametrizedInput(
    val withStatus: EventStatus?,
    val createdAfter: String?
) : GraphQLParametrizedInput {

    override fun deepCopy(): EventPropertyParentParametrizedInput {
        return EventPropertyParentParametrizedInput(
            this.withStatus,
            this.createdAfter
        )
    
    }

    override fun toString(): String {
        val joiner = StringJoiner(", ", "( ", " )")
        if (withStatus != null) {
            joiner.add("withStatus: " + GraphQLRequestSerializer.getEntry(withStatus))
        }
        if (createdAfter != null) {
            joiner.add("createdAfter: " + GraphQLRequestSerializer.getEntry(createdAfter))
        }
        return joiner.toString()
    }
}
